### DISCUSSION TITLE
Using VisualStudioCode-PlatformIO

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#45] [General]
Under the PROJECT TASKS Folder (left Side), there are two folders - Default and firebeetle2.

1. I assume I should use the tasks under the firebeetle2/General sub-folder.
2. Build and Upload are the only two tasks needed to be executed.
3. Build resulting log is:

- Building in release mode
- Calculating size .pio/build/firebeetle2/firmware.elf
   text        data        bss      dec           hex       filename
1075821  270296   26881 1372998  14f346 .pio/build/firebeetle2/firmware.elf
========================================= [SUCCESS] Took 12.42 seconds 

 I assume [SUCCESS} is what we are looking for

It looks like the Upload task does a Build task, then the firmware upload.  Is this correct?

**Comment by alex-code** [comment]
I think if there's only config for one board then the tasks in **Default** and **firebeetle2** do the same thing.

You'd only need `Build` if you made changes and want to check it compiles.
`Upload` will always rebuild (if there's any changes) then upload (if the build succeeds).

**Comment by daveclifford** [comment]
You've clarified the difference between Default and FireBeetle2 folders. 

I will use the FireBeetle2 folder only.

Thanks for the response.

Dave
